staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -90,6 +90,22 @@ type Config struct {
 	// StorageObjectCountTracker is used to keep track of the total
 	// number of objects in the storage per resource.
 	StorageObjectCountTracker flowcontrolrequest.StorageObjectCountTracker
+
+	// AllowUnsafeCorruptObjectDeletion, if enabled, will make an attempt to
+	// perform the normal deletion flow, but if either of the below occurs:
+	// a) the data (associated with the resource being deleted) retrieved
+	// from the storage failed to transform properly (eg. decryption failure)
+	// b) the data (associated with the resource being deleted) failed to
+	// decode properly (eg. corrupt data)
+	// it will disregard these errors, bypass the finalzer constraints,
+	// deletion hook(s) and go ahead with the deletion flow.
+	//
+	// WARNING: This will break the cluster if the resource has dependencies
+	// Use only if you REALLY know what you are doing.
+	// WARNING: Vendor(s) will most likely consider using this option to be
+	// in violation of the support of their product.
+	// The default value is false, and the user must opt in to enable it.
+	AllowUnsafeCorruptObjectDeletion bool
```

Comment:
- In the future, when the feature gate is removed, this can be used as a knob to enable/disable the feature.
- this also gives us the ability to tweak this feature per resource, in case we decide that we will never enable this feature for certain types.

I can remove this and just use  `utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AllowUnsafeMalformedObjectDeletion)`, thoughts @sttts @stlaz ?

Reply:
No strong opinion. But I tend to agree with @stlaz. We can make it configurable when we see the need.